addons/api/addon/models/session.js

```diff
@@ -75,16 +75,30 @@ class SessionCredential {
   get secrets() {
     if (this.#payloadSecret?.decoded) {
       const secretJSON = this.#payloadSecret.decoded;
-      return Object.keys(secretJSON).map(
-        (key) => new SessionCredential.SecretItem(key, secretJSON[key]),
-      );
+      return this.extractSecrets(secretJSON);
     } else {
       // decode from base64
       const decodedString = atob(this.#payloadSecret.raw);
       return [new SessionCredential.SecretItem('secret', decodedString)];
     }
   }
 
+  /**
+   * Extracts secrets from the payload secret JSON object.
+   * @param {object} secretJSON - The payload secret JSON object.
+   * @returns {SessionCredential.SecretItem[]} - The array of secret items.
+   */
+  extractSecrets(secretJSON) {
+    // If `decoded` contains a `data` object and if that object has a `username` and `password`,
+    // then we need to return those as key-value pairs
+    // TODO: will there be other labels besides `username` and `password`?
```

Comment:
we discussed this offline, but I'll add the gist for context-  we will be displaying everything that we get from `data` in a key/value copyable format(similar to how we show static creds types such as username/password), and we will not show metadata as that is not useful to the user. 

right now, we will show with the parent level key  

```
data.nested: 'more data'  --> copyable 
other: 'wow' --> copyable 
```

let me your thoughts on this approach.

Reply:
Are we making this change only for vault cred store? I think we should keep the logic the same for static cred since it can be any arbitrary values but it seems like vault returns back it as a specific format with the `data` and `metadata`.

Are you able to return any arbitrary nested object from vault? I'm curious if we need to do the `data.nested` and recurse through the object (and whether we should)

Reply:
the logic is the same for static and vault creds. The JSON blob from static credentials will also now be displayed in key/value format. Pretty much everything sent to decoded will be displayed in key/value format, except `metadata`  as we decided against displaying that to the user. 

I removed the path prepend logic and now show it as a flat single object key and value, so it will be like the below example. I also added screenshots, where I use different cred types from both static and vault and how they would look with this new change

for ex: if the user passes this as JSON static credential,
```
{
  data: {
    nested: "more data"
  },
  other: "wow"
}
```
we will see
```
nested: 'more data'  --> copyable 
other: 'wow' --> copyable 
```

Reply:
I think my point was that static cred stores also have arbitrary json types, should we be unfurling those recursively? Should we just limit this to vault types and only look at the `data` field? What happens with collisions?

Reply:
oops, yes you are right, I will limit it to just vault. I misunderstood when I was looking at static JSON creds, I thought we didn't support key/value and only displayed the JSON. 

 what sort of collisions are you thinking? can you give me an example?